src/thunderbots/software/ai/navigator/obstacle/velocity_obstacle.cpp

```diff
@@ -0,0 +1,49 @@
+#include "ai/navigator/obstacle/velocity_obstacle.h"
+Polygon VelocityObstacle::getBoundaryPolygon(const Robot& robot,
+                                             double robot_radius_scaling,
+                                             double velocity_projection_scaling)
+{
+    double radius = ROBOT_MAX_RADIUS_METERS * robot_radius_scaling;
+    Vector velocity_projection =
+        robot.velocity().norm(velocity_projection_scaling * robot.velocity().len());
+
+    if (velocity_projection.len() > radius)
+    {
+        Vector velocity_direction_norm_radius = robot.velocity().norm(radius);
+        return Polygon(
+            {// sides of robot
+             robot.position() + velocity_direction_norm_radius.rotate(Angle::quarter()),
+             robot.position() +
+                 velocity_direction_norm_radius.rotate(Angle::threeQuarter()),
+             // velocity projections
+             robot.position() + velocity_direction_norm_radius.rotate(Angle::quarter()) +
+                 velocity_projection,
+             robot.position() +
+                 velocity_direction_norm_radius.rotate(Angle::threeQuarter()) +
+                 velocity_projection,
+             // back of robot
+             robot.position() +
+                 velocity_direction_norm_radius.rotate(Angle::ofDegrees(150)),
+             robot.position() +
+                 velocity_direction_norm_radius.rotate(Angle::ofDegrees(210))});
```

Comment:
Could we make all of these descriptively named variables, and then construct the Polygon from those? As it stands now I have a lot of trouble reading this. Similarily for the polygon below.

Reply:
To agree with Gareth, it would be more legible if you created some constants that are explained individually. When everything is clumped together it isn't as easy to write descriptive comments